lstchain/tools/lstchain_create_dl3_file.py

```diff
@@ -151,14 +164,48 @@ def setup(self):
 
     def start(self):
 
-        self.data = read_data_dl2_to_QTable(str(self.input_dl2))
-        ## To reduce the table columns further, add a selection of columns to be read and used.
-        self.effective_time, self.elapsed_time = get_effective_time(self.data)
-        self.run_number = run_info_from_filename(self.input_dl2)[1]
+        if not self.source_dep:
+            self.data = read_data_dl2_to_QTable(str(self.input_dl2))
+            ## To reduce the table columns further, add a selection of columns to be read and used.
+            self.effective_time, self.elapsed_time = get_effective_time(self.data)
+            self.run_number = run_info_from_filename(self.input_dl2)[1]
+
+            self.data = self.event_sel.filter_cut(self.data)
+            self.data = self.fixed_cuts.gh_cut(self.data)
+            self.data = add_icrs_position_params(self.data, self.source_pos)
 
-        self.data = self.event_sel.filter_cut(self.data)
-        self.data = self.fixed_cuts.gh_cut(self.data)
-        self.data = add_icrs_position_params(self.data, self.source_pos)
+        else:
+            
+            srcdep_pos_keys = get_srcdep_index_keys(self.input_dl2)
+
+            for i, srcdep_pos in enumerate(srcdep_pos_keys):
+                data_temp = read_data_dl2_to_QTable(str(self.input_dl2), srcdep_pos=srcdep_pos)
+
+                if i==0:
+                    self.effective_time, self.elapsed_time = get_effective_time(data_temp)
+                    self.run_number = run_info_from_filename(self.input_dl2)[1]
+
+                data_temp = self.event_sel.filter_cut(data_temp)
+                data_temp = self.fixed_cuts.gh_cut(data_temp)
+                data_temp = self.fixed_cuts.alpha_cut(data_temp)
+                data_temp = add_icrs_position_params(data_temp, self.source_pos)
+
+                # set expected source positions as reco positions
+                time = data_temp['dragon_time']
+                obstime = Time(time, scale='utc', format='unix')
+                expected_src_x = data_temp['expected_src_x'] * u.m
+                expected_src_y = data_temp['expected_src_y'] * u.m
+                focal = 28 * u.m
+                pointing_alt = data_temp['pointing_alt']
+                pointing_az  = data_temp['pointing_az']
+                expected_src_altaz = camera_to_altaz(expected_src_x, expected_src_y, focal, pointing_alt, pointing_az, obstime=obstime)
+                data_temp["reco_alt"] = expected_src_altaz.alt
+                data_temp["reco_az"]  = expected_src_altaz.az
```

Comment:
Probably better to create a separate function like add_icrs_position_params in lstchain/irf/hdu_table.py, so as to keep the Tool simple